runtime/src/bank.rs

```diff
@@ -1695,10 +1698,12 @@ impl Bank {
             freeze_started: AtomicBool::new(false),
             cost_tracker: RwLock::new(CostTracker::default()),
             sysvar_cache: RwLock::new(SysvarCache::default()),
-            accounts_data_len: AtomicU64::new(parent.load_accounts_data_len()),
+            accounts_data_len: AtomicU64::new(accounts_data_len),
             fee_structure: parent.fee_structure.clone(),
         };
 
+        new.set_cost_tracker_account_data_size_limit(accounts_data_len);
```

Comment:
same, instead of `set...`, would prefer to pass `accounts_data_len` to cost_tracker constructor.